### ISSUE TITLE
Google Drive "List Files" operation only processes the first item

### ISSUE THREAD

**Issue by Yann-J** [n8n-io/n8n#3414]
**Describe the bug**

The "List files and folders" node seems to only execute the search query on the first incoming item.

This is confirmed by looking at [the code](https://github.com/n8n-io/n8n/blob/master/packages/nodes-base/nodes/Google/Drive/GoogleDrive.node.ts#L2372), which immediately returns the list instead of appending it.

I understand the expected behaviour is ambiguous, since this node returns a list (and would therefore need to output a list of lists).

To illustrate why the current behaviour is a problem, here's my use case:

* I'm polling some data, including binary content, through a REST API returning a list of items.
* My workflow will determine an appropriate naming convention (folder and file name) to upload the binaries on Google Drive
* I therefore need, for each item, to first check if my target folder exists in Google Drive (which I do thanks to this node) and, if not already existing, will create it.

This means for such a workflow, the ideal behaviour is to return exactly one output item for each input item, with either the id of the existing folder, or an empty value.

**To Reproduce**
1. Create a workflow with a node producing a list of items (e.g. a `Function` node returning a list of items with a `FolderName` property)
2. Add a Google Drive node with the "search files and folders", and using a search query for each of the folder names returned by the previous node (e.g. `mimeType = 'application/vnd.google-apps.folder' and trashed = false and name = '{{$json["FolderName"]}}'`), and a maximum number of results per query of `1`.
3. Execute the workflow and check that the Google Drive node only returned an entry for the first query.

Here's a sample workflow to help reproduce (tweak the values and credentials accordingly).

```
{
  "name": "Test G Drive",
  "nodes": [
    {
      "parameters": {},
      "name": "Start",
      "type": "n8n-nodes-base.start",
      "typeVersion": 1,
      "position": [
        240,
        300
      ]
    },
    {
      "parameters": {
        "functionCode": "return [\n  {json:{FolderName: 'Farmer Pics 2'}},\n  {json:{FolderName: 'Folder 1'}},\n  {json:{FolderName: 'Map Tiles'}},\n];"
      },
      "name": "Function",
      "type": "n8n-nodes-base.function",
      "typeVersion": 1,
      "position": [
        460,
        300
      ]
    },
    {
      "parameters": {
        "operation": "list",
        "useQueryString": true,
        "queryString": "=mimeType = 'application/vnd.google-apps.folder' and trashed = false and name = '{{$json[\"FolderName\"]}}'",
        "limit": 1,
        "options": {}
      },
      "name": "Google Drive",
      "type": "n8n-nodes-base.googleDrive",
      "typeVersion": 1,
      "position": [
        680,
        300
      ],
      "credentials": {
        "googleApi": {
          "id": "6",
          "name": "Google Service Account - ito-googledrive"
        }
      }
    }
  ],
  "connections": {
    "Start": {
      "main": [
        [
          {
            "node": "Function",
            "type": "main",
            "index": 0
          }
        ]
      ]
    },
    "Function": {
      "main": [
        [
          {
            "node": "Google Drive",
            "type": "main",
            "index": 0
          }
        ]
      ]
    }
  },
  "active": false,
  "settings": {},
  "id": 21
}
```

**Expected behavior**

I think the most reasonable behaviour in this situation is to return a concatenated list of all the search results for all the input items.

**Environment (please complete the following information):**
 - n8n Version `0.177.0` (official Docker image)
 - Database system `postgres`
 - Operation mode `queue`

**Comment by Joffcom** [comment]
Nice find @Yann-J, Seems to only happen when using a query string as well.

**Comment by Yann-J** [comment]
OK the fix seemed easy enough so I gave it a try in #3419

**Comment by janober** [comment]
Fix got merged with #3525 and will be released with the next version. Thanks a lot!

**Comment by janober** [comment]
Fix got released with `n8n@0.186.0`
